src/installer/__main__.py

```diff
@@ -0,0 +1,83 @@
+"""Installer CLI."""
+
+import argparse
+import os.path
+import re
+import sys
+import sysconfig
+from typing import Dict, Optional, Sequence
+
+import installer
+import installer.destinations
+import installer.sources
+import installer.utils
+
+
+def main_parser() -> argparse.ArgumentParser:
+    """Construct the main parser."""
+    parser = argparse.ArgumentParser()
+    parser.add_argument("wheel", type=str, help="wheel file to install")
+    parser.add_argument(
+        "--destdir",
+        "-d",
+        metavar="path",
+        type=str,
+        help="destination directory (prefix to prepend to each file)",
+    )
+    parser.add_argument(
+        "--compile-bytecode",
+        action="append",
+        metavar="level",
+        type=int,
+        choices=[0, 1, 2],
+        help="generate bytecode for the specified optimization level(s) (default=0, 1)",
+    )
+    parser.add_argument(
+        "--no-compile-bytecode",
+        action="store_true",
+        help="don't generate bytecode for installed modules",
+    )
+    return parser
+
+
+def get_scheme_dict(distribution_name: str) -> Dict[str, str]:
+    """Calculate the scheme dictionary for the current Python environment."""
+    scheme_dict = sysconfig.get_paths()
+
+    # calculate 'headers' path, not currently in sysconfig - see
+    # https://bugs.python.org/issue44445. This is based on what distutils does.
+    # For new wheels, the name we get from the wheel filename should already
+    # be normalised (based on PEP 503, but with _ instead of -), but this is
+    # not the case for many existing wheels, so normalise it here.
+    normed_name = re.sub(r"[-_.]+", "_", distribution_name).lower()
```

Comment:
This shouldn't be necessary, right? The distribution name is extracted from the filename of the wheel.

Reply:
Sorry, I totally missed the comment!

Reply:
We've only just agreed that distribution names in wheel filenames should be normalised like this, and it's not yet implemented in wheel. So we have to handle wheel filenames where it's not normalised.

Of course, we could use the original distribution name instead of the normalised one. But I think that requires more changes, because that has to be read from the metadata in the wheel.

Reply:
Thanks. For anybody else who wasn't aware, the discussion is at https://discuss.python.org/t/revisiting-distribution-name-normalization/12348/15.

Reply:
(For some reason I can't collapse my own review comments, feel free to.)